realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionImpl.java

```diff
@@ -539,6 +548,7 @@ protected E convertRowToObject(UncheckedRow row) {
         if (className != null) {
             return new OrderedRealmCollectionSnapshot<E>(realm, collection, className);
         } else {
+            assert classSpec != null;
```

Comment:
Generally, we haven't used java asserts anywhere before. So I would be a bit hesitant about doing it here? What is the reason for adding them?